File: Task-03/primes.cpp
#include <iostream>
#include <vector>

std::vector<int> find_primes(int n)
{
    std::vector<bool> arr(n - 1, true);

    for (int p = 2; p * p <= n; p++)
    {
        if (arr[p - 2])
        {
            for (int i = p * p; i <= n; i += p)
            {
                arr[i - 2] = false;
            }
        }
    }

    std::vector<int> primes;
    for (int i = 2; i <= n; i++)
    {
        if (arr[i - 2])
        {
            primes.push_back(i);
        }
    }

    return primes;
}

int main()
{
    int num;
    printf("Enter a number:");
    scanf("%d", &num);
    std::vector<int> primeNumbers = find_primes(num);
    printf("prime numbers upto %d are:",num);
    for (int prime : primeNumbers)
    {
        printf("%d ",prime);
    }

    return 0;
}

File: Task-03/primes.rs
fn find_primes(n: usize) -> Vec<usize> {
    let mut arr = vec![true; n -1];
    
    let mut p = 2;
    while p * p <= n {
        if arr[p-2] {
            let mut i = p * p;
            while i <= n {
                arr[i-2] = false;
                i += p;
            }
        }
        p += 1;
    }
    let mut primes = Vec::new();
    for i in 2..=n {
        if arr[i-2] {
            primes.push(i);
        }
    }

    primes
}

fn main() {
    let num = 50; 
    let primes = find_primes(num);

    println!("Prime numbers up to {} are: {:?}", limit, primes);
}
